dvc/output/base.py

```diff
@@ -373,14 +378,12 @@ def checkout(
                 filter_info=filter_info,
                 **kwargs,
             )
-            if self.isexec:
-                st = os.stat(self.path_info.fspath)
-                os.chmod(self.path_info.fspath, st.st_mode | stat.S_IEXEC)
             return res
         except CheckoutError:
             if allow_missing or self.checkpoint:
                 return None
             raise
+        self.set_exec()
```

Comment:
@efiop  This line will probably not execute because of `return` on line 381

Reply:
Yep, put it there by accident. Thanks!